### PR TITLE
nodejs: update to 11.6.0.

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

**Comment by Kistelini** [comment]
You'll need this patch: https://gitweb.gentoo.org/repo/gentoo.git/tree/net-libs/nodejs/files/nodejs-99999999-llhttp.patch

**Comment by Cogitri** [comment]
Thanks.

**Comment by Cogitri** [comment]
> You'll need this patch: https://gitweb.gentoo.org/repo/gentoo.git/tree/net-libs/nodejs/files/nodejs-99999999-llhttp.patch

Have decided against using this, I just packaged llhttp properly.

File: common/shlibs [hunk 1]
```diff
@@ -3380,3 +3380,4 @@ librtasevent.so.2 librtas-2.0.2_1
 libbroker.so.0 bro-2.6.1_1
 libcaf_core.so.0.16.2 bro-2.6.1_1
 libcaf_openssl.so.0.16.2 bro-2.6.1_1
+libllhttp.so.1 llhttp-1.0.1_1
```
Comment (maxice8) [inline common/shlibs]: missing newline at the end of file

File: srcpkgs/llhttp-devel [hunk 2]
```diff
@@ -0,0 +1 @@
+llhttp/
```
Comment (maxice8) [inline srcpkgs/llhttp-devel]: no leading slash on symlink please.

File: srcpkgs/llhttp/files/Makefile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# Makefile template for shared library
+
+CC ?= gcc
```
Comment (maxice8) [inline srcpkgs/llhttp/files/Makefile]: problably not set it and let us set it ?

File: srcpkgs/nodejs/template [hunk 4]
```diff
@@ -61,6 +61,14 @@ do_configure() {
 		$(vopt_if nghttp2 --shared-nghttp2) \
 		$(vopt_if cares --shared-cares) ${_args}
 }
+
+post_configure() {
+	# Fix linking against llhttp
+	sed 's/http_parser/& -lllhttp/' \
+		-i  out/cctest.target.mk \
```
Comment (maxice8) [inline srcpkgs/nodejs/template]: one has 2 whitespace, another has 1

File: srcpkgs/llhttp/files/Makefile (hunk 3 above)
Comment (Cogitri) [reply srcpkgs/llhttp/files/Makefile]: Uh? `?=` means default, if you set CC it'll be overruled

File: srcpkgs/nodejs/template [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
 homepage="https://nodejs.org/"
 distfiles="${homepage}/dist/v${version}/node-v${version}.tar.gz"
-checksum=dbe467e3dabb6854fcb0cd96e04082268cb1e313ce97a4b7100b2ed152b0a0ab
+checksum=39ef4f1866f75786baff5959439483fafdc99d3ee3a0568a13cc635d64cf5e0b
 
 build_options="ssl libuv http_parser icu nghttp2 cares"
 desc_option_ssl="Enable shared libressl"
 desc_option_libuv="Enable shared libuv"
-desc_option_http_parser="Enable shared http-parser"
+desc_option_http_parser="Enable shared http-parser&llhttp"
```
Comment (maxice8) [inline srcpkgs/nodejs/template]: i'd rather have `http-parser and llhttp`

File: srcpkgs/llhttp/files/Makefile (hunk 3 above)
Comment (maxice8) [reply srcpkgs/llhttp/files/Makefile]: ah ok

**Pull request merged**